nodeup/pkg/model/kubelet.go

```diff
@@ -438,6 +439,23 @@ func (b *KubeletBuilder) buildKubeletConfigSpec() (*kops.KubeletConfigSpec, erro
 		c.BootstrapKubeconfig = ""
 	}
 
+	if b.Cluster.Spec.Networking != nil && b.Cluster.Spec.Networking.AmazonVPC != nil {
+		instanceType, err := awsup.GetMachineTypeInfo(b.InstanceGroup.Spec.MachineType)
+		if err != nil {
+			return c, err
+		}
+
+		maxPods := int32(instanceType.MaxPods)
+		c.MaxPods = &maxPods
+		if b.InstanceGroup.Spec.Kubelet != nil {
+			if b.InstanceGroup.Spec.Kubelet.MaxPods == nil {
```

Comment:
I'm also wondering if we should only set MaxPods _lower_, so the default is around 100 I believe.  The problem is that e.g. polling all the containers is pretty slow.

OTOH I'm also open to just setting it everywhere, because it's the bigger instances anyway that allow more than 100 VPC addresses.